imxrt1060-hal/src/gpio.rs

```diff
@@ -160,12 +167,26 @@ where
         // The input / output direction. When switching across fast / normal, keep the
         // same direction.
         copy_bits!(GDIR, self.mask());
+
+        // Interrupt configuration is preserved when switching fast / normal.
+        if <P as Pin>::Offset::USIZE < 16 {
+            copy_bits!(ICR1, self.mask());
+        } else {
+            copy_bits!(ICR2, self.mask());
```

Comment:
We'll need to use ICR masks here. ICR fields are two bits wide. To hold all the interrupt configurations for up to 32 GPIOs, we need 64 bits. That's why there's two 32-bit ICR registers.

```suggestion
            copy_bits!(ICR1, 0b11 << self.icr_offset());
        } else {
            copy_bits!(ICR2, 0b11 << self.icr_offset());
```